WhitespaceAfter should support LITERAL_TRY token

https://checkstyle.org/config_whitespace.html#WhitespaceAfter

I have downloaded the latest cli from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

**How it works Now:**

```bash
/var/tmp $ javac TryStatement.java
/var/tmp $

/var/tmp $ cat config.xml
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="WhitespaceAfter">
            <property name="tokens" value="LITERAL_TRY"/>
        </module>
    </module>
</module>

/var/tmp $ cat TryStatement.java
import java.io.IOException;
import java.io.InputStream;

public class TryStatement {
    public static void main(String[] args) throws IOException {
        try(InputStream ignored = System.in) {

        }
    }
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.1-all.jar -c config.xml TryStatement.java
com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - Token "LITERAL_TRY" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:476)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:403)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:330)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:189)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:126)
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Token "LITERAL_TRY" was not found in Acceptable tokens list in check com.puppycrawl.tools.checkstyle.checks.whitespace.WhitespaceAfterCheck
        at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:223)
        at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:133)
        at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:201)
        at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:471)
        ... 5 more
Checkstyle ends with 1 errors.
```

**Is your feature request related to a problem? Please describe.**
It would be nice if checkstyle could detect missing space after `try` keyword.

**Describe the solution you'd like**
WhitespaceAfter could support new literal type - `LITERAL_TRY`

**Additional context**
Google Java Style Guide requires whitespace separation of `(` from any reserved words
See https://google.github.io/styleguide/javaguide.html#s4.6.2-horizontal-whitespace
>1. Separating any reserved word, such as `if`, `for` or `catch`, from an open parenthesis (`(`) that follows it on that line

#8122 seems related. This feature request is part of it.

--- a/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
+++ b/src/it/java/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/WhitespaceAfterTest.java
@@ -56,6 +56,7 @@ public class WhitespaceAfterTest extends AbstractGoogleModuleTestSupport {
             "22:28: " + getCheckMessage(clazz, message, "..."),
             "23:26: " + getCheckMessage(clazz, message, "->"),
             "24:9: " + getCheckMessage(clazz, message, "switch"),
+            "31:9: " + getCheckMessage(clazz, message, "try"),
         };
         final Configuration checkConfig = getModuleConfig("WhitespaceAfter");
         final String filePath = getPath("InputWhitespaceAfterBad.java");

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterBad.java
@@ -26,4 +26,8 @@ public class InputWhitespaceAfterBad {
                 break;
         }
     }
+
+    public void check4() throws java.io.IOException {
+        try(java.io.InputStream ignored = System.in;) { } // warn
+    }
 }

--- a/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
+++ b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputWhitespaceAfterGood.java
@@ -26,4 +26,9 @@ public class InputWhitespaceAfterGood {
                 break;
         }
     }
+
+    public void check4() throws java.io.IOException {
+        try (java.io.InputStream ignored = System.in) {}
+        try {} catch(Exception e){}
+    }
 }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -60,6 +60,8 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  * ELLIPSIS</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
  * LITERAL_SWITCH</a>,
+ * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
+ * LITERAL_TRY</a>,
  * <a href="https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
  * LAMBDA</a>.
  * </li>
@@ -84,6 +86,12 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  *
  *      for (;;){}               // OK
  *      for(;;){}                // violation, space after 'for' is required
+ *
+ *      try (InputStream ignored = System.in) {} // OK
+ *      try(InputStream ignored = System.in) {} // violation ''try' is not followed by whitespace'
+ *
+ *      try {} catch (Exception e){} // OK
+ *      try{} catch (Exception e){} // violation ''try' is not followed by whitespace'
  *      }
  * </pre>
  * <p>
@@ -158,6 +166,7 @@ public class WhitespaceAfterCheck
             TokenTypes.DO_WHILE,
             TokenTypes.ELLIPSIS,
             TokenTypes.LITERAL_SWITCH,
+            TokenTypes.LITERAL_TRY,
             TokenTypes.LAMBDA,
         };
     }

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheck.java
@@ -92,7 +92,7 @@ import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
  *
  *      try {} catch (Exception e){} // OK
  *      try{} catch (Exception e){} // violation ''try' is not followed by whitespace'
- *      }
+ *  }
  * </pre>
  * <p>
  * To configure the check for whitespace only after COMMA and SEMI tokens:

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/whitespace/WhitespaceAfterCheck.xml
@@ -11,7 +11,7 @@
  EmptyForIteratorPad&lt;/a&gt; to validate empty for iterators.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LAMBDA"
+            <property default-value="COMMA,SEMI,TYPECAST,LITERAL_IF,LITERAL_ELSE,LITERAL_WHILE,LITERAL_DO,LITERAL_FOR,DO_WHILE,ELLIPSIS,LITERAL_SWITCH,LITERAL_TRY,LAMBDA"
                       name="tokens"
                       type="java.lang.String[]"
                       validation-type="tokenSet">

--- a/src/main/resources/google_checks.xml
+++ b/src/main/resources/google_checks.xml
@@ -106,7 +106,7 @@
       <property name="tokens"
                value="COMMA, SEMI, TYPECAST, LITERAL_IF, LITERAL_ELSE,
                     LITERAL_WHILE, LITERAL_DO, LITERAL_FOR, DO_WHILE, ELLIPSIS,
-                    LITERAL_SWITCH, LAMBDA"/>
+                    LITERAL_SWITCH, LITERAL_TRY, LAMBDA"/>
     </module>
     <module name="WhitespaceAround">
       <property name="allowEmptyConstructors" value="true"/>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WhitespaceAfterCheckTest.java
@@ -149,6 +149,17 @@ public class WhitespaceAfterCheckTest
     }
 
     @Test
+    public void testLiteralTry() throws Exception {
+        final String[] expected = {
+            "20:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "try"),
+            "24:9: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, "try"),
+        };
+        verifyWithInlineConfigParser(
+            getPath("InputWhitespaceAfterLiteralTry.java"),
+            expected);
+    }
+
+    @Test
     public void testEmptyForIterator() throws Exception {
         final String[] expected = {
             "18:30: " + getCheckMessage(MSG_WS_NOT_FOLLOWED, ";"),

--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralTry.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/whitespace/whitespaceafter/InputWhitespaceAfterLiteralTry.java
@@ -1,0 +1,26 @@
+/*
+WhitespaceAfter
+tokens = LITERAL_TRY
+
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.whitespace.whitespaceafter;
+
+import java.io.IOException;
+import java.io.InputStream;
+
+public class InputWhitespaceAfterLiteralTry {
+    public static void main(String[] args) throws IOException {
+        try (InputStream ignored = System.in) {} // OK
+
+        try // OK
+            (InputStream ignored = System.in) {}
+
+        try(InputStream ignored = System.in) {} // violation ''try' is not followed by whitespace'
+
+        try {}catch (Exception e){} // OK
+
+        try{}catch (Exception e){} // violation ''try' is not followed by whitespace'
+    }
+}

--- a/src/xdocs/config_whitespace.xml
+++ b/src/xdocs/config_whitespace.xml
@@ -2906,6 +2906,8 @@ class Bar {
                   ELLIPSIS</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
                   LITERAL_SWITCH</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
+                LITERAL_TRY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
                   LAMBDA</a>
                   .
@@ -2934,6 +2936,8 @@ class Bar {
                   ELLIPSIS</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_SWITCH">
                   LITERAL_SWITCH</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LITERAL_TRY">
+                LITERAL_TRY</a>
                 , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#LAMBDA">
                   LAMBDA</a>
                   .
@@ -2966,6 +2970,12 @@ public void myTest() {
 
     for (;;){}               // OK
     for(;;){}                // violation, space after 'for' is required
+
+    try (InputStream ignored = System.in) {} // OK
+    try(InputStream ignored = System.in) {} // violation ''try' is not followed by whitespace'
+
+    try {} catch (Exception e){} // OK
+    try{} catch (Exception e){} // violation ''try' is not followed by whitespace'
 }
           </pre>
         </div>

